main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+)
+
+func Generator(ctx context.Context, ch chan<- int64, fn func(int64)) {
+	defer close(ch)
+	var I int64 = 1
```

Comment:
Локальные переменные, особенно счетчики, принято писать маленькими буквами
```suggestion
	var i int64 = 1
```